src/realm/collection_parent.hpp

```diff
@@ -71,40 +71,44 @@ enum class UpdateStatus {
     /// current state.
     NoChange,
 };
-
 struct PathElement {
     union {
         std::string string_val;
         int64_t int_val;
     };
-    enum Type { string, integer } m_type;
+    enum Type { column, key, index } m_type;
```

Comment:
When is `PAthElement` used for storing a ck? I did not see it being used in this PR...

Reply:
Obj::add_index and Obj::get_path